### ISSUE TITLE
Bug: Private channels with token gated access disappear

### ISSUE THREAD

**Issue by chair28980** [status-im/status-react#20999]
Related to https://github.com/status-im/status-desktop/issues/16067

# Bug Report

## Problem 
Issue is present on both mobile and desktop.

#### Expected behavior
Private channels with token gated access are always visible and accessible. 

#### Actual behavior
Token-gated "private" channels are not always present/accessible.

I expect to have access to "private-general" channel via nft possession in my linked address.

### Reproduction
Not predictably reproducible, requires monitoring channel list when opening the app.
1. Open mobile app
2. Inspect channel listing in community

### Additional Information

- Status version: nightly 
- Operating System: iOS

On mobile device, without expected access to 'private-general'.
![image](https://github.com/user-attachments/assets/e1be9c5f-bd48-4b58-8366-6fce0d73d116)

**Comment by churik** [comment]
I don't think it is fixed to be hones, as I see this issue come up from time to time - so token-gated channels might be locked despite the user still have access to them. And as far as I understood it is a deeper problem with token-gated communities itself and it doesn't depend on the client.

I'll leave it open, as it is definitely the case still
